client/src/components/Chat.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { useAuth } from '../AuthContext';
+import { database } from '../firebaseConfig';
+import { ref, push, onValue, serverTimestamp } from 'firebase/database';
+
+interface Admin {
+  id: string;
+  username: string;
+}
+
+interface Message {
+  sender_id: string;
+  message: string;
+  timestamp: number;
+}
+
+function Chat() {
+  const { user } = useAuth();
+  const [message, setMessage] = useState<string>('');
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [admins, setAdmins] = useState<Admin[]>([]);
+  const [selectedAdmin, setSelectedAdmin] = useState<Admin | null>(null);
+
+  useEffect(() => {
+    const fetchAdmins = async () => {
+      const response = await axios.get('/api/auth/online-admins', { withCredentials: true });
```

Comment:
What does this route do ? Just out of curiosity

Reply:
The route to fetch the shelter that's online. I have corrected the API URL as well.